src/pe/data_directories.rs

```diff
@@ -1,20 +1,114 @@
 use crate::error;
-use scroll::{Pread, Pwrite, SizeWith};
+use core::ops::Deref;
+use scroll::{
+    ctx::{self, TryFromCtx},
+    Endian, Pread, Pwrite, SizeWith,
+};
 
 #[repr(C)]
 #[derive(Debug, PartialEq, Copy, Clone, Default, Pread, Pwrite, SizeWith)]
-pub struct DataDirectory {
+pub struct DataDirectoryInner {
```

Comment:
this is a breaking change, just fyi

Reply:
Yeah, I'm not happy about that

Reply:
why does the offset have to be tracked in this manner, just for the pwrite implementation or?

Reply:
I fixed it back to a non-breaking change. We can always revisit if we need to make it breaking later on.